Preprocessing/DataExtractionFromWiki/EpisodeInfoProcessor.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import json
+from pathlib import Path
+import re
+import sys
+from typing import (
+    Any,
+    Dict,
+    Optional,
+)
+
+from Preprocessing.ErrorHandlingLogger import ErrorHandlingLogger
+from Preprocessing.utils import setup_logger
+
+
+class EpisodeInfoProcessor:
+    def __init__(self, base_path: str, episode_info_path: str, output_path: str):
+        self.base_path = Path(base_path)
+        self.episode_info_path = Path(episode_info_path)
+        self.output_path = Path(output_path)
+        self.logger = ErrorHandlingLogger(
+            class_name=self.__class__.__name__,
+            logger=setup_logger(self.__class__.__name__),
+        )
+
+    def run(self) -> int:
+        try:
+            if not self.base_path.is_dir():
+                self.logger.error(f"Invalid base path: {self.base_path}")
+                return 2
+
+            episode_info = self.load_episode_info()
+
+            for file_path in self.base_path.rglob("*.json"):
+                self.process_file(file_path, episode_info)
```

Comment:
```suggestion
            if self.base_path.is_dir():
                episode_info = self.load_episode_info()
                for file_path in self.base_path.rglob("*.json"):
                    self.process_file(file_path, episode_info)
            else:
                self.logger.error(f"Invalid base path: {self.base_path}")
```

ctrl shift f na return 1 / return 2 zrob w folderze preprocessing, bo mialo tego nie byc XD